grakn-test/src/test/java/ai/grakn/test/engine/lock/ZookeeperLockTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.test.engine.lock;
+
+import ai.grakn.engine.lock.ZookeeperLock;
+import ai.grakn.engine.tasks.manager.ZookeeperConnection;
+import ai.grakn.exception.EngineStorageException;
+import ai.grakn.test.EngineContext;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class ZookeeperLockTest {
```

Comment:
I would change this to `LockTest` and test against the interface. Then by using Junit `Theories` you can make sure that your different locks behave in the same manner. This could be reserved for a different PR though but I think is a good change to make.